Add user changeable option for scmi protocol

biwenli opened the pull request:
Add user changeable option for scmi protocol

iuliana-prodan commented on drivers/firmware/scmi/Kconfig:
```diff
@@ -64,6 +64,12 @@ config ARM_SCMI_TRANSPORT_INIT_PRIORITY
 
+config SCMI_CHAN_SEM_TIMEOUT_USEC
+	int "SCMI channel semaphore timeout(unit: usec)"
+	default 500
```
How did you get this number? 
Why is different from one target to another?

iuliana-prodan commented on soc/nxp/imx/imx9/imx943/Kconfig.defconfig.mimx94398.m33:
```diff
@@ -30,4 +30,7 @@ config ETH_NXP_IMX_MSGINTR
 config CACHE_MANAGEMENT
 	default y
 
+config SCMI_CHAN_SEM_TIMEOUT_USEC
```
The commit message says: "to fix a bug that failed to setup pins" - this is too generic. 
Please provide more details and context. It appears to be a hardware-related issue, with additional steps required for the i.MX943?
How did you identify the bug?
Why is a delay 100 times longer than the default necessary?

LaurentiuM1234 commented on drivers/firmware/scmi/Kconfig:
```diff
@@ -64,6 +64,12 @@ config ARM_SCMI_TRANSPORT_INIT_PRIORITY
 	help
 	  SCMI transport driver device initialization priority.
 
+config SCMI_CHAN_SEM_TIMEOUT_USEC
```
keep the entries sorted alphabetically

LaurentiuM1234 commented on drivers/firmware/scmi/Kconfig:
```diff
@@ -64,6 +64,12 @@ config ARM_SCMI_TRANSPORT_INIT_PRIORITY
+	int "SCMI channel semaphore timeout(unit: usec)"
+	default 500
+	help
+	  SCMI channel semaphore timeout(unit: usec).
```
how about something more descriptive like: `Maximum number of microseconds to wait for a reply from the platform`?

LaurentiuM1234 commented on soc/nxp/imx/imx9/imx943/Kconfig.defconfig.mimx94398.m33:
```diff
@@ -30,4 +30,7 @@ config ETH_NXP_IMX_MSGINTR
 	default y
 
+config SCMI_CHAN_SEM_TIMEOUT_USEC
+	default 50000
```
oh wow, this is quite a jump from 500 us to 50000. I'm curious about the context of the problem.

LaurentiuM1234 reviewed (commented):
I agree with the SCMI-related changes. Different platforms have different communication latencies. However, as @iuliana-prodan pointed out, it would be good to have a more detailed description of the problem (for the MX943-related changes)

biwenli commented:
> I agree with the SCMI-related changes. Different platforms have different communication latencies. However, as @iuliana-prodan pointed out, it would be good to have a more detailed description of the problem (for the MX943-related changes)

sure, np.

biwenli committed:
drivers: firmware: scmi: add CONFIG_ARM_SCMI_CHAN_SEM_TIMEOUT_USEC

Add option CONFIG_ARM_SCMI_CHAN_SEM_TIMEOUT_USEC
for users to change default timeout value.
Different platform has different delay time when
scmi agent get scmi response from scmi platform.

Signed-off-by: Biwen Li <biwen.li@nxp.com>
